### DISCUSSION TITLE
Support for `--root-path` argument

### DISCUSSION THREAD

**Discussion by GokulNC** [coder/code-server#4626] [Q&A] [answered]
I am looking to add a prefix path for code-server instead of serving it via `/`.
This is to ensure that I can route VS code via my regular port 80 website, but with a prefix (like abc.com/code ) by linking it to my proxy-server to map it to code-server's port and path.

Basically, I am looking for something like this:
https://fastapi.tiangolo.com/advanced/behind-a-proxy/

If there's already a way to achieve this, please let me know. Thanks!

**Comment by GokulNC** [comment]
As of now, without this, I am getting `404: Not Found` error when visiting abc.com/code since it redirects to abc.com/login

(My reverse proxy-server strips the `/code` prefix before sending it the code-server's port)

**Comment by GokulNC** [comment]
OK. So if I explicitly go to `/code/login`, it works after logging-in as intended.
So is it possible to fix this initial redirection issue?

---

Also, only `/code/` works, `/code` does not work (after login).

**Comment by jsjoeio** [comment, marked as answer]
Hmmm...I believe this already works with something like Caddy. Here's an example:
1. create a `Caddyfile`
```Caddyfile
http://localhost:8082/code/* {
  uri strip_prefix /code
  reverse_proxy 127.0.0.1:8080
}
```
2. start code-server with `code-server`
3. `caddy run` to start Caddy
4. Go to `http://localhost:8082/code/login` and then login
5. Access code-server like normal

Is this what you want or am I missing something? 

> fix this initial redirection issue

I wasn't sure about this part but maybe that's something we can add to the Caddyfile

**Reply by GokulNC** [reply]
Thanks. I use `traefik` to achieve the same.

The issue seems to be with `traefik`, sorry.
This is the redirection issue I was talking about:
https://github.com/traefik/traefik/issues/5159

---

For other people looking to solve this, as a temporary solution, on my `traefik` server's `routes.toml` file, I have added the following middleware:

```
    [http.middlewares.code-slash-suffix.redirectRegex]
      regex = "/code$"
      replacement = "/code/"
```

**Reply by jsjoeio** [reply]
Ah, good to know! Happy you found a workaround at least and hoping it gets fixed in `traefik` soon 🤞
